File: tests/testdata/languages/csharp/basic.cs
using System;

namespace Acme.Auth;

public class AuthService {
    private readonly string authHeader = "X-Auth-Header";
    public string Token { get; init; }

    public AuthService(string token) {
        Token = token;
    }

    public bool Login(string user) {
        return !string.IsNullOrWhiteSpace(user);
    }
}

File: tests/testdata/languages/zig/basic.zig
const std = @import("std");

pub const Client = struct {
    token: []const u8,

    pub fn login(self: *Client, user: []const u8) bool {
        return user.len > 0 and self.token.len > 0;
    }
};

pub fn buildClient(token: []const u8) Client {
    return .{ .token = token };
}
